modules/sdk-coin-avaxp/src/lib/transaction.ts

```diff
@@ -146,7 +155,41 @@ export class Transaction extends BaseTransaction {
    */
   get signablePayload(): Buffer {
     const txbuff = this._avaxpTransaction.toBuffer();
-    return createHash.default('sha256').update(txbuff).digest();
+    return utils.sha256(txbuff);
+  }
+
+  get id(): string {
+    return utils.cb58Encode(utils.sha256(utils.cb58Decode(this.toBroadcastFormat())) as any as BufferAvax);
```

Comment:
why is this cast needed? Seems unsafe

Reply:
Avalanche's Buffer and Node's Buffer are similar. Both extend from UintArray. 
I try to avoid the Buffer.from that convert one in other. I accidentally left it here.  
I'm still interested in a way to avoid those conversion. I didn't find a way.